aiida_common_workflows/workflows/relax/cp2k/generator.py

```diff
@@ -190,36 +199,39 @@ def _construct_builder(self, **kwargs) -> engine.ProcessBuilder:
         kpoints_distance = protocol_dict.pop('kpoints_distance', None)
         kpoints = self._get_kpoints(kpoints_distance, structure, reference_workchain)
         mesh, _ = kpoints.get_kpoints_mesh()
-        if mesh != [1, 1, 1]:
+        if 'sirius' in protocol:
+            parameters['FORCE_EVAL']['PW_DFT']['PARAMETERS']['NGRIDK'] = f'{mesh[0]} {mesh[1]} {mesh[2]}'
+        elif mesh != [1, 1, 1]:
             builder.cp2k.kpoints = kpoints
 
         magnetization_tags = None
 
         # Metal or insulator.
-        # If metal then add smearing, unoccupied orbitals, and employ diagonalization.
-        if electronic_type == ElectronicType.METAL:
-            parameters['FORCE_EVAL']['DFT']['SCF']['SMEAR'] = {
-                '_': 'ON',
-                'METHOD': 'FERMI_DIRAC',
-                'ELECTRONIC_TEMPERATURE': '[K] 710.5',
-            }
-            parameters['FORCE_EVAL']['DFT']['SCF']['DIAGONALIZATION'] = {
-                'EPS_ADAPT': '1',
-            }
-            parameters['FORCE_EVAL']['DFT']['SCF']['MIXING'] = {
-                'METHOD': 'BROYDEN_MIXING',
-                'ALPHA': '0.1',
-                'BETA': '1.5',
-            }
-            parameters['FORCE_EVAL']['DFT']['SCF']['ADDED_MOS'] = 20
-            parameters['FORCE_EVAL']['DFT']['SCF']['CHOLESKY'] = 'OFF'
-
-        # If insulator then employ OT.
-        elif electronic_type == ElectronicType.INSULATOR:
-            parameters['FORCE_EVAL']['DFT']['SCF']['OT'] = {
-                'PRECONDITIONER': 'FULL_SINGLE_INVERSE',
-                'MINIMIZER': 'CG',
-            }
+        if 'sirius' in protocol:
```

Comment:
@hmhoseini was it on purpose that you deleted only this section and not the section that deals with magnetisation?

Reply:
Parameters related to "SMEARING" are given in PW_DFT section. 
The rest is not necessary.

Reply:
Parameters related to "SMEARING" are given in PW_DFT section. 
The rest is not necessary.

Reply:
> Parameters related to "SMEARING" are given in PW_DFT section.

In the current implementation, smearing is always enabled. Can you change it so it is switched on for metals and switched off for insulators?

> The rest is not necessary.

I do not understand this sentence. Do you mean the magnetisation section stays the same when using sirius?

Reply:
> > Parameters related to "SMEARING" are given in PW_DFT section.
> 
> In the current implementation, smearing is always enabled. Can you change it so it is switched on for metals and switched off for insulators?

No! 
https://manual.cp2k.org/trunk/CP2K_INPUT/FORCE_EVAL/PW_DFT/PARAMETERS.html#FORCE_EVAL.PW_DFT.PARAMETERS.SMEARING

 
> 
> > The rest is not necessary.
> 
> I do not understand this sentence. Do you mean the magnetisation section stays the same when using sirius?

I mean, OT section is irrelevant for PW-DFT calculations.

Reply:
> No!
> https://manual.cp2k.org/trunk/CP2K_INPUT/FORCE_EVAL/PW_DFT/PARAMETERS.html#FORCE_EVAL.PW_DFT.PARAMETERS.SMEARING

It is very hard for me to understand the answer. Could you please be slightly more explicit so that I don't need to interpret it? 

My understanding is that `sirius` doesn't provide an option to disable smearing, is that correct?

> I mean, OT section is irrelevant for PW-DFT calculations.

Ok, clear. But the other question still remains unanswered. There is a part of the code that handles starting magnetisation:

https://github.com/aiidateam/aiida-common-workflows/blob/cf4dbfd33997dad3b0325ebf00e6263e50d3bf6f/aiida_common_workflows/workflows/relax/cp2k/generator_sirius.py#L184-L198

Will it remain the same for sirius?

Reply:
> > No!
> > https://manual.cp2k.org/trunk/CP2K_INPUT/FORCE_EVAL/PW_DFT/PARAMETERS.html#FORCE_EVAL.PW_DFT.PARAMETERS.SMEARING
> 
> It is very hard for me to understand the answer. Could you please be slightly more explicit so that I don't need to interpret it?
> 
> My understanding is that `sirius` doesn't provide an option to disable smearing, is that correct?
> 
It is correct.  I though it is clear from the document I sent. And this is the reason I changed the generator for PW-DFT calculations. 

> > I mean, OT section is irrelevant for PW-DFT calculations.
> 
> Ok, clear. But the other question still remains unanswered. There is a part of the code that handles starting magnetisation:
> 
> https://github.com/aiidateam/aiida-common-workflows/blob/cf4dbfd33997dad3b0325ebf00e6263e50d3bf6f/aiida_common_workflows/workflows/relax/cp2k/generator_sirius.py#L184-L198
> 
> Will it remain the same for sirius?
I do not know if PW spin-polarized calculation is implemented in CP2K. If it is implemented, the only section that control spin-polarized calculation is DFT/UKS. Therefore, I did not changed this section in sirius generator.